fypp-gfortran.py

```diff
@@ -0,0 +1,59 @@
+import sys
+import subprocess
+from pathlib import Path
+
+# Read the system arguments.
+args = sys.argv[1:]
+
+# Set the output directory.
+output_dir = Path("build" , "preprocessed_files")
+
+# If output directory does not exist, create it.
+if not output_dir.exists():
+    output_dir.mkdir(parents=True)
+
+# Get the filenames with .fypp extension and convert to string.
+filenames = [str(f) for f in Path(".").glob("**/*.fypp")]
```

Comment:
You should filter the filename from `sys.argv` rather than globbing the whole source tree.

Reply:
I had tried using `filenames = [x for x in args if x.endswith(".fypp")]` instead of using `filenames = [str(f) for f in Path(".").glob("**/*.fypp")]`. But, the filenames list came as empty and in the build/preprocessed_files, I had no generated files which denoted that in` sys.argv[1:] ` or `args`, there are no file paths with `.fypp` extension